numpy/random/_common.pyx

```diff
@@ -444,7 +453,18 @@ cdef int check_constraint(double val, object name, constraint_type cons) except
             raise ValueError("{0} < 0 or {0} is NaN".format(name))
         elif not (val <= LEGACY_POISSON_LAM_MAX):
             raise ValueError(name + " value too large")
+    elif cons == CONS_NEGATIVE_BINOMIAL_POISSON:
+        if not (val <= POISSON_LAM_MAX):
+            raise ValueError("{0} is invalid, one value is too large or the other too small".format(name))
+
+    return 0
 
+cdef int check_constraint_pair(double val0, double val1, object name_0, object name_1, constraint_type cons) except -1:
+    cdef double max_lam
+    if cons == CONS_NEGATIVE_BINOMIAL_POISSON:
+        max_lam = (1 - val1) / val1 * (val0 + 10 * np.sqrt(np.array(val0).astype(np.double)))
```

Comment:
Why cast double values to array?  This will be quite slow. Just use `sqrt` from libc on doubles.